client-src/elements/chromedash-metadata.ts

```diff
@@ -1,39 +1,42 @@
 import {LitElement, css, html} from 'lit';
+import {customElement, property, state} from 'lit/decorators.js';
 import {ifDefined} from 'lit/directives/if-defined.js';
 import {SHARED_STYLES} from '../css/shared-css.js';
+import {Channels} from '../js-src/cs-client.js';
 
-class ChromedashMetadata extends LitElement {
-  static get properties() {
-    return {
-      implStatuses: {type: Array}, // Read in chromedash-featurelist.
-      status: {attribute: false}, // Read in chromedash-featurelist.
-      selected: {attribute: false}, // Directly edited in /templates/features.html
-      _className: {attribute: false},
-      _fetchError: {attribute: false},
-      _channels: {attribute: false},
-      _versions: {attribute: false},
-    };
-  }
+interface InplStatus {
+  key: number;
+  val: string;
+}
 
-  constructor() {
-    super();
-    this.implStatuses = [];
-    this.status = {
-      NO_ACTIVE_DEV: 1,
-      PROPOSED: 2,
-      IN_DEVELOPMENT: 3,
-      BEHIND_A_FLAG: 4,
-      ENABLED_BY_DEFAULT: 5,
-      DEPRECATED: 6,
-      REMOVED: 7,
-      ORIGINTRIAL: 8,
-      INTERVENTION: 9,
-      ON_HOLD: 10,
-      NO_LONGER_PURSUING: 1000,
-    };
-    this._channels = {};
-    this._versions = [];
-  }
+@customElement('chromedash-metadata')
+class ChromedashMetadata extends LitElement {
+  @property({type: Array})
+  implStatuses: InplStatus[] = [];
+  @property({type: Object, attribute: false})
+  status = {
+    NO_ACTIVE_DEV: 1,
+    PROPOSED: 2,
+    IN_DEVELOPMENT: 3,
+    BEHIND_A_FLAG: 4,
+    ENABLED_BY_DEFAULT: 5,
+    DEPRECATED: 6,
+    REMOVED: 7,
+    ORIGINTRIAL: 8,
+    INTERVENTION: 9,
+    ON_HOLD: 10,
+    NO_LONGER_PURSUING: 1000,
+  };
+  @property({type: String, attribute: false})
+  selected!: string;
+  @state()
+  _className: 'canaryisdev' | 'betaisdev' | undefined;
+  @state()
+  _fetchError!: boolean;
+  @state()
+  _channels: Channels = {};
+  @state()
+  _versions!: (string | number)[];
```

Comment:
@jyasskin i'm not sure if this is correct but based on the assignment below it seems to be correct. Could you help double check this? thanks